### PR TITLE
Data-Driven Keyboard Conversions: 0-9

### PR DESCRIPTION
**Pull request by noroadsleft**
## Description

Converts configuration definitions and rules to data-driven where applicable. Renames `info.json` to `keyboard.json` where migrations have taken place.

- keyboards with names beginning with `[0-9]`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by noroadsleft** [comment]
Affects:

```
- `0xc7/61key`
- `0xcb/1337`
- `0xcb/static`
- `1upkeyboards/1up60hse`
- `1upkeyboards/1up60hte`
- `1upkeyboards/1up60rgb`
- `1upkeyboards/1upocarina`
- `1upkeyboards/1upsuper16v3`
- `1upkeyboards/pi60`
- `1upkeyboards/pi60_hse`
- `1upkeyboards/pi60_rgb`
- `1upkeyboards/pi60_rgb_v2`
- `1upkeyboards/super16`
- `1upkeyboards/super16v2`
- `1upkeyboards/sweet16/v1`
- `1upkeyboards/sweet16v2/kb2040`
- `1upkeyboards/sweet16v2/pro_micro`
- `25keys/aleth42`
- `25keys/cassette42`
- `25keys/zinc`
- `2key2crawl`
- `40percentclub/25`
- `40percentclub/4pack`
- `40percentclub/4x4`
- `40percentclub/5x5`
- `40percentclub/6lit`
- `40percentclub/foobar`
- `40percentclub/half_n_half`
- `40percentclub/i75`
- `40percentclub/luddite`
- `40percentclub/mf68`
- `40percentclub/nein`
- `40percentclub/nori`
- `40percentclub/polyandry`
- `40percentclub/tomato`
- `40percentclub/ut47`
- `45_ats`
- `4pplet/aekiso60/rev_b`
- `4pplet/eagle_viper_rep`
- `4pplet/steezy60`
- `4pplet/unextended_std/rev_a`
- `4pplet/waffling60`
- `4pplet/waffling80`
- `4pplet/yakiimo/rev_a`
- `8pack`
```

File: keyboards/1upkeyboards/super16/super16.c
Comment (fauxpark) [inline keyboards/1upkeyboards/super16/super16.c:1]: Some of these `g_led_configs` don't make sense. I would prefer if they were not migrated yet, until they can all be verified.

File: keyboards/40percentclub/ut47/info.json [hunk 1]
```diff
@@ -10,6 +10,24 @@
   },
   "processor": "atmega32u4",
   "bootloader": "caterina",
+  "matrix_pins": {
+    "rows": ["D1", "D0", "D4", "C6"],
+    "cols": ["D7", "E6", "B4", "B5", "B6", "B2", "B3", "B1", "F7", "F6", "F5", "F4"]
+  },
+  "diode_direction": "COL2ROW",
```
Comment (fauxpark) [inline keyboards/40percentclub/ut47/info.json:17]: This board uses a custom matrix, these should not be migrated.

File: keyboards/1upkeyboards/super16/super16.c
Comment (noroadsleft) [reply keyboards/1upkeyboards/super16/super16.c:1]: https://github.com/noroadsleft/qmk_firmware/compare/61f56998d1..0481165c7b

**Comment by dunk2k** [comment]
Quick check, more so my curiosity: would this be a good opportunity to start replacing `processor` and `bootloader` with `development_board`, where applicable?

**Comment by noroadsleft** [comment]
@dunk2k, 

I think that would be worth doing, but it's out of scope for this PR (/ series of PRs) - I'm only replacing like for like.

**Comment by dunk2k** [comment]
> @dunk2k,
> 
> I think that would be worth doing, but it's out of scope for this PR (/ series of PRs) - I'm only replacing like for like.

Just what I was thinking 😆
A nice to do but unfortunately would add complexity to what's going to be a rather hefty series of PRs.

File: keyboards/40percentclub/4pack/keyboard.json [hunk 2]
```diff
@@ -23,6 +23,12 @@
         "mousekey": false,
         "nkro": false
     },
+    "qmk": {
+        "locking": {
+            "enabled": "True",
+            "resync": "True"
```
Comment (dunk2k) [inline keyboards/40percentclub/4pack/keyboard.json:29]: CI Build fix
```suggestion
            "enabled": true,
            "resync": true
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/40percentclub/4pack/keyboard.json (hunk 2 above)
Comment (noroadsleft) [reply keyboards/40percentclub/4pack/keyboard.json]: https://github.com/noroadsleft/qmk_firmware/compare/3dfe833164..f68348d8f3

**Review by dunk2k — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by zvecr — APPROVED** [review]

**Pull request merged**